### PR TITLE
Run required workflow jobs when dependencies are skipped

### LINKED ISSUE #6319: "data" and "po" changes don't trigger builds in the workflow
tothxa*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/4682)*
*Created on Mon Jan 15 22:12:40 CET 2024 by **Tóth András (tothxa)***

---

**Describe the bug**
Changes in the `data` directory should trigger all jobs to run through `testsuite` and `installer`. The `changes` action identifies them, but the dispatcher doesn't start them. In the same way, `po` changes (translation updates) should trigger w*ndows and macos builds through `installer`, but don't.

**To reproduce**
See:
 - `po`: https://github.com/widelands/widelands/actions/runs/7523633110
 - `data`: https://github.com/widelands/widelands/actions/runs/7533923802

**Expected behavior**
The appropriate workflow jobs should have been run.

**Version:**
 - OS: github actions
 - Widelands Version: https://codeberg.org/wl/widelands/commit/fcb77638ae5f64ef560e1621d650ff7d12eaa118

### PR DESCRIPTION
**Pull request by tothxa**
**Type of change**
Bugfix

**Issue(s) closed**
Fixes #6319

**Possible regressions**
workflow runs

### CODE REVIEW

**Comment by tothxa** [comment]
Whew... seems to work now. See workflow runs of last 3 commits.

File: .github/workflows/pre-build_checks.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        required: true
+        type: string
+      codecheck:
+        required: true
+        type: string
+
+jobs:
+  clang_tidy:
+    if: ${{ inputs.clang_tidy != 'false' }}
+    uses: ./.github/workflows/build_clang_tidy.yaml
```
Comment (matthiakl) [inline .github/workflows/pre-build_checks.yaml:15]: Is there a special reason to check `!false` instead of `true`?

Comment (tothxa) [reply .github/workflows/pre-build_checks.yaml:15]: Yes, to fail towards running instead of skipping if there's any problem with the passed input.

**Review by matthiakl — APPROVED** [review]
Ok, I hope this will be the last fix required for the new workflow structure.

**Comment by tothxa** [comment]
Thank you.

> Ok, I hope this will be the last fix required for the new workflow structure.

I guess we all do...

@bunnybot merge

**Pull request closed without merging**